embedded-graphics/src/pixel_iterator.rs

```diff
@@ -0,0 +1,29 @@
+//! Pixel iterator
+
+use crate::drawable::Pixel;
+use crate::pixelcolor::PixelColor;
+use crate::primitives::rectangle::Rectangle;
+
+/// Pixel iterator trait
+pub trait PixelIterator<C>
+where
+    C: PixelColor,
+{
+    /// TODO: Doc
+    type Iter: Iterator<Item = Pixel<C>>;
+
+    /// TODO: Doc
+    fn pixels(self) -> Self::Iter;
+}
+
+/// Sparse pixel iterator
+pub trait SparsePixelIterator<C>
+where
+    C: PixelColor,
+{
+    ///  TODO: Doc
+    type Iter: Iterator<Item = Option<C>>;
+
+    /// TODO: Doc
+    fn sparse_pixels(self) -> (Rectangle, Self::Iter);
+}
```

Comment:
Would this work?

```suggestion
pub trait SparsePixelIterator<C>
where
    C: PixelColor,
{
    ///  TODO: Doc
    type Iter: Iterator<Item = Option<C>> + Dimensions;

    /// TODO: Doc
    fn sparse_pixels(self) -> Self::Iter;
}
```

Reply:
Ah yeah that's a much better idea. We'd have to `impl Dimensions` for all the styled iterators which isn't difficult, just a bit of boilerplate.